lib/koala/api/graph_batch_api.rb

```diff
@@ -53,7 +53,15 @@ def execute(http_options = {})
           end
 
           original_api.graph_call("/", args, "post", http_options) do |response|
-            raise bad_response if response.nil?
+            raise bad_response("Facebook returned an empty body") if response.nil?
+
+            # when http_component is set we receive Koala::Http_service response object
+            # from graph_call so this needs to be parsed
+            # as generate_results method handles only JSON response
+            if http_options[:http_component]
+              response = JSON.load(response.body)
```

Comment:
Could you handle JSONError and add a spec for it. If fb sends a messed up body we will end up with a json error instead of a koala error.

And from experience, fb loves to send html error pages or null instead of well formed json

Reply:
We have added the code change to handle the JSON error and added test coverage as well.